src/wp-includes/post-thumbnail-template.php

```diff
@@ -176,6 +176,19 @@ function get_the_post_thumbnail( $post = null, $size = 'post-thumbnail', $attr =
 			update_post_thumbnail_cache();
 		}
 
+		// Get the 'loading' attribute value to use as default, taking precedence over the default from
+		// `wp_get_attachment_image()`.
+		$loading = wp_get_loading_attr_default( 'the_post_thumbnail' );
+
+		// Add the default to the given attributes unless they already include a 'loading' directive.
+		if ( empty( $attr ) ) {
+			$attr = array( 'loading' => $loading );
+		} elseif ( is_array( $attr ) && ! array_key_exists( 'loading', $attr ) ) {
+			$attr['loading'] = $loading;
+		} elseif ( is_string( $attr ) && ! preg_match( '/(\?|&)loading=', $attr ) ) {
```

Comment:
The `?` here doesn't seem right. The `$attr` here is a query string, not a URL with a query string attached to it. So I think this is rather needing to see if `loading=` occurs at the beginning of the query string or as a key somewhere inside:

```suggestion
		} elseif ( is_string( $attr ) && ! preg_match( '/(^|&)loading=', $attr ) ) {
```

Reply:
Good catch, updated!